### PR TITLE
makefile: synthesis does not need versions.txt

### PR DESCRIPTION
**Pull request by oharboe**

### CODE REVIEW

**Comment by oharboe** [comment]
@maliberty unrelated failures

File: flow/Makefile [hunk 1]
```diff
@@ -442,8 +442,7 @@ $(WRAPPED_LIBS):
 # |____/ |_| |_| \_| |_| |_| |_|_____|____/___|____/
 #
 .PHONY: synth
-synth: versions.txt \
```
Comment (maliberty) [inline flow/Makefile:445]: Something needs to trigger the generation of this file.  Removing it here means it wont' be generated by default

Comment (oharboe) [reply flow/Makefile:445]: What is it used for?

Comment (maliberty) [reply flow/Makefile:445]: It makes it easy to know what versions of tools someone is using.

Comment (oharboe) [reply flow/Makefile:445]: in make issue?

Comment (oharboe) [reply flow/Makefile:445]: > It makes it easy to know what versions of tools someone is using.

Except it doesn't do that... Only if the "synth" target is invoked explicitly is this file written... so `make issue` is likely to have the *wrong* version.

If I run  `make nuke && make DESIGN_CONFIG=designs/asap7/gcd/config.mk floorplan`, then no versions.txt is written.

Comment (maliberty) [reply flow/Makefile:445]: It is a quite old construct so perhaps it is no longer attached to the right target.   Attaching it to make issue seems reasonable.

Comment (oharboe) [reply flow/Makefile:445]: Then this PR is good to go...

```
$ ls versions.txt
ls: cannot access 'versions.txt': No such file or directory
$ make DESIGN_CONFIG=designs/asap7/gcd/config.mk floorplan_issue
[deleted]
$ ls versions.txt
versions.txt
$ cat versions.txt
Yosys 0.13+15 (git sha1 bc027b2ca, gcc 12.3.0-1ubuntu1~23.04 -fPIC -Os)
openroad v2.0-11432-gead3236e5
KLayout 0.28.11
$ echo >versions.txt blah
$ cat versions.txt
blah
$ make DESIGN_CONFIG=designs/asap7/gcd/config.mk floorplan_issue
[deleted]
$ cat versions.txt
Yosys 0.13+15 (git sha1 bc027b2ca, gcc 12.3.0-1ubuntu1~23.04 -fPIC -Os)
openroad v2.0-11432-gead3236e5
KLayout 0.28.11
$ 
```

**Review by maliberty — APPROVED** [review]

**Pull request merged**